source/env_tutorial/bsuite.rst

```diff
@@ -0,0 +1,190 @@
+
+.. _Actions space-1:
+Actions space
+--------
+
+-  The action space is a discrete space of size N, (where N can vary according to the environment), datatype is \ ``int``\ and input is a python integer value（or a np array of dimension 0 such as \ ``np.array(1)``\ to input action 1）.
+
+-  For example, in the Deep Sea environment, N is equal to 2, thus action values ranges from 0 to 1. For their specific meaning you can refer to the following list.
+
+   -  0：LEFT.
+
+   -  1：RIGHT.
+
+.. _Rewards space-1:
+Rewards space
+--------
+
+-  Rewards are assigned according to the rules of the environments, rewards are usually a \ ``float``\ value.
+
+.. _Other-1:
+Others
+----
+
+-  Environments terminate once they have reached their maximum number of steps or is encountered a failure state. All environments have a fixed number of maximum steps, but not all environments have a failure state.
+
+Key facts
+========
+
+1. Each environment contains several configurations to make it gradually more challenging.
+
+2. Discrete actions space.
+
+3. Each environment is designed to test a particular propriety of RL policies, including: generalization, exploration, credit assignment, scaling, noise, memory.
+
+4. Rewards scale can vary a lot.
+
+.. _Other-3:
+Others
+====
+
+Using bsuite in 'OpenAI Gym' format
+----------
+
+Our implementation uses the bsuite Gym wrapper to make the bsuite codebase run under the OpenAI Gym interface. Hence, ``gym`` needs to be installed to make bsuite work properly.
+
+.. code:: python
+
+   import bsuite
+   from bsuite.utils import gym_wrapper
+   env = bsuite.load_and_record_to_csv('catch/0', results_dir='/path/to/results')
+   gym_env = gym_wrapper.GymFromDMEnv(env)
+
+About configurations
+--------
+
+Configurations are designed to increase the level of difficulty of an environment. For example, in a 5-armed bandit environment, configurations are used to regulate the level of noise to perturb the rewards.
+Given a specific environment, all possible configurations can be visualized with the following code snippet.
+
+.. code:: python
+
+   from bsuite import sweep
+   print('List bsuite_id for "bandit_noise" experiment:')
+   print(sweep.BANDIT_NOISE)
+
+.. image:: ./images/bsuite_config.png
+   :align: center
+
+Using DI-engine, you can create a bsuite environment simply with the name of your desired configuration.
+
+.. code:: python
+
+   from easydict import EasyDict
+   from dizoo.bsuite.envs import BSuiteEnv
+   cfg = {'env': 'memory_len/0'}
+   cfg = EasyDict(cfg)
+   memory_len_env = BSuiteEnv(cfg)
+
+
+DI-zoo runnable code
+====================
+
+The full training configuration can be found on `github
+link <https://github.com/opendilab/DI-engine/tree/main/dizoo/bsuite/config/serial>`__
+. Following, we show an example of configuration for the file, ``memory_len_0_dqn_config.py``\, you can run the demo with the following code：
```

Comment:
In the following part, we show an example of configuration for the file, ``memory_len_0_dqn_config.py``\, you can run the demo with the following code：